### PR TITLE
Expand Policy to include all protection options

### PR DESCRIPTION
**Pull request by fejta**
/assign @stevekuznetsov @cjwagner @BenTheElder 

This adds all the missing fields and deprecates the existing ones.

Let me know if you want me to further break this down (actually don't just review please, although I guess I could decouple deprecating the old fields with adding the new ones if necessary ;-) )

### CODE REVIEW

File: prow/cmd/branchprotector/protect_test.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ import (
 	"k8s.io/test-infra/prow/github"
 
 	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/util/diff"
```
Comment (stevekuznetsov) [inline prow/cmd/branchprotector/protect_test.go]: nit: this import block should be above the `test-infra` one

File: prow/cmd/branchprotector/request.go [hunk 2]
```diff
@@ -17,6 +17,8 @@ limitations under the License.
 package main
 
 import (
+	"log"
+
 	branchprotection "k8s.io/test-infra/prow/config"
```
Comment (stevekuznetsov) [inline prow/cmd/branchprotector/request.go]: nit: this import should be with the other `test-infra` one in a block

File: prow/cmd/branchprotector/request.go [hunk 3]
```diff
@@ -25,29 +27,75 @@ import (
...
+
+// makeReviews renders review policy into the corresponding GitHub api object.
+//
+// Returns nil if the policy is nil, or approvals is nil or 0.
+func makeReviews(rp *branchprotection.ReviewPolicy) *github.RequiredPullRequestReviews {
+	switch {
+	case rp == nil:
+		return nil
+	case rp.Approvals == nil:
+		log.Println("WARNING: required_pull_request_reviews policy does not specify required_approving_review_count, disabling")
```
Comment (stevekuznetsov) [inline prow/cmd/branchprotector/request.go]: Should be `logrus`

File: prow/config/branch_protection.go [hunk 4]
```diff
@@ -17,16 +17,87 @@ limitations under the License.
 package config
 
 import (
+	"errors"
 	"fmt"
+	"log"
```
Comment (stevekuznetsov) [inline prow/config/branch_protection.go]: `logrus`

File: prow/cmd/branchprotector/protect_test.go (hunk 1 above)
Comment (fejta) [reply prow/cmd/branchprotector/protect_test.go]: Done

File: prow/cmd/branchprotector/request.go (hunk 2 above)
Comment (fejta) [reply prow/cmd/branchprotector/request.go]: Done

File: prow/cmd/branchprotector/protect_test.go (hunk 1 above)
Comment (fejta) [reply prow/cmd/branchprotector/protect_test.go]: Done

File: prow/cmd/branchprotector/request.go (hunk 2 above)
Comment (fejta) [reply prow/cmd/branchprotector/request.go]: Done

File: prow/cmd/branchprotector/request.go (hunk 3 above)
Comment (fejta) [reply prow/cmd/branchprotector/request.go]: Done

File: prow/config/branch_protection.go (hunk 4 above)
Comment (fejta) [reply prow/config/branch_protection.go]: Done

File: prow/cmd/branchprotector/protect_test.go [hunk 5]
```diff
@@ -24,11 +24,11 @@ import (
 	"strings"
 	"testing"
 
+	"k8s.io/apimachinery/pkg/util/diff"
```
Comment (stevekuznetsov) [inline prow/cmd/branchprotector/protect_test.go]: err sorry as I was not clear, generally we try to:

```go
import (
    "standard"
    "library"
    "packages"

    "github.com/third"
    "github.com/third/party"
    "github.com/third/party/imports"

    "github.com/kubernetes/test-infra/imports"
    "github.com/kubernetes/test-infra/from"
    "github.com/kubernetes/test-infra/this"
    "github.com/kubernetes/test-infra/repo"
)
```

in three sections

**Pull request merged**